core/src/bundle_stage/result.rs

```diff
@@ -1,37 +1,39 @@
 use {
-    crate::{
-        bundle_stage::bundle_account_locker::BundleAccountLockerError, tip_manager::TipPaymentError,
-    },
+    crate::{bundle_stage::bundle_account_locker::BundleAccountLockerError, tip_manager::TipError},
     anchor_lang::error::Error,
+    serde::{Deserialize, Serialize},
     solana_bundle::bundle_execution::LoadAndExecuteBundleError,
     solana_poh::poh_recorder::PohRecorderError,
     thiserror::Error,
 };
 
 pub type BundleExecutionResult<T> = Result<T, BundleExecutionError>;
 
-#[derive(Error, Debug, Clone)]
+// NOTE: The variants are aliased for backwards compatibility. Fields were renamed in 1.16 causing issues when clients use the 1.14 sdk.
+#[derive(Error, Debug, Clone, Serialize, Deserialize)]
 pub enum BundleExecutionError {
-    #[error("PoH record error: {0}")]
-    PohRecordError(#[from] PohRecorderError),
-
-    #[error("Bank is done processing")]
-    BankProcessingDone,
-
-    #[error("Execution error: {0}")]
-    ExecutionError(#[from] LoadAndExecuteBundleError),
+    #[error("The bank has hit the max allotted time for processing transactions")]
+    BankProcessingTimeLimitReached,
 
     #[error("The bundle exceeds the cost model")]
     ExceedsCostModel,
 
-    #[error("Tip error {0}")]
-    TipError(#[from] TipPaymentError),
+    #[serde(alias = "TransactionFailure")]
+    #[error("Runtime error while executing the bundle: {0}")]
+    ExecutionError(#[from] LoadAndExecuteBundleError),
```

Comment:
should we just rename?

Reply:
wfm

Reply:
just worried about thrashing (that i already introduced)

Reply:
i think breaking API change is fine; v1.16.15 needs to be cut soon and v1.14 -> v1.16 had breaking api changes

Reply:
can tell users to upgrade

Reply:
cool